components/kyma-environment-broker/internal/process/deprovisioning/event_hub_step.go

```diff
@@ -0,0 +1,130 @@
+package deprovisioning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	processazure "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+type DeprovisionAzureEventHubStep struct {
+	OperationManager *process.DeprovisionOperationManager
+	instanceStorage  storage.Instances
+	processazure.EventHub
+}
+
+func NewDeprovisionAzureEventHubStep(os storage.Operations, is storage.Instances, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) DeprovisionAzureEventHubStep {
+	return DeprovisionAzureEventHubStep{
+		OperationManager: process.NewDeprovisionOperationManager(os),
+		instanceStorage:  is,
+		EventHub: processazure.EventHub{
+			HyperscalerProvider: hyperscalerProvider,
+			AccountProvider:     accountProvider,
+			Context:             ctx,
+		},
+	}
+}
+
+var _ Step = (*DeprovisionAzureEventHubStep)(nil)
+
+func (s DeprovisionAzureEventHubStep) Name() string {
+	return "Deprovision Azure Event Hubs"
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
+	hypType := hyperscaler.Azure
+
+	instance, err := getInstance(s.instanceStorage, operation, log)
+	switch err.(type) {
+	case instanceNotFoundError:
+		return s.OperationManager.OperationSucceeded(operation, "instance already deprovisioned")
+	case instanceGetError:
+		return operation, 1 * time.Second, nil
+	}
+
+	// parse provisioning parameters (instance being nil is protected by the previous switch case)
+	pp, err := instance.GetProvisioningParameters()
```

Comment:
if you need only globalaccountid you can get it directly from the instance

Reply:
@piotrmiskiewicz Thanks for your review.
I believe it's needed also for getting the azure config. @nachtmaar right?

Reply:
Correct. The region is required in order to create the azure client. See [here](https://github.com/kyma-incubator/compass/blob/0ac13cfb71fb47d895c4539ba0936aa573ff8833/components/kyma-environment-broker/internal/hyperscaler/azure/env.go#L50)